Security doc updates

Amruta-Ranade opened the pull request:
Closes #3023 and #4052

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/docs/4420)
<!-- Reviewable:end -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/331390f1b407143c53c1df3c899a2c2769c0b6ae/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/44b3d5ec620d6f2032ac640c2337263748d1386b/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/e4652a4f1693a6ce52584caf18d80e173daad1d1/

sploiselle commented on v19.1/encryption.md:
```diff
@@ -21,8 +21,7 @@ Encryption at Rest provides transparent encryption of a node's data on the local
 
-Encryption at Rest allows encryption of all files on disk using [AES](https://en.wikipedia.org/wiki/Advanced_Encryption_Standard) in [counter mode](https://en.wikipedia.org/wiki/Block_cipher_mode_of_operation#Counter_(CTR)), with all key
-sizes allowed.
+Encryption at Rest allows encryption of all files on disk using [AES](https://en.wikipedia.org/wiki/Advanced_Encryption_Standard) in [counter mode](https://en.wikipedia.org/wiki/Block_cipher_mode_of_operation#Counter_(CTR)), with all AES-Standard key sizes allowed.
```
Encryption at rest seems like it belongs in its own page

Amruta-Ranade replied:
I consolidated the encryption docs as part of the security project I did last quarter. Making a separate doc for encryption at rest would warrant an independent discussion. Making a note of it.

sploiselle commented on v19.1/encryption.md:
```diff
@@ -131,21 +117,67 @@ Generating a key file can be done using the `cockroach` CLI:
+
+#### Step 3. Start a node with encryption
 
 Encryption is configured at node start time using the `--enterprise-encryption` command line flag.
```
s/Encryption/Encryption at rest?

sploiselle commented on v19.1/encryption.md:
```diff
@@ -165,16 +197,16 @@ Starting a node for the first time using AES-128 encryption can be done using:
+#### Step 4. Check the encryption status
 
-Encryption status can be seen on the node's stores report, reachable through: `http(s)://nodeaddress:8080/#/reports/stores/local` (or replace `local` with the node ID). For example, if you are running a [local cluster](secure-a-cluster.html), you can see the node's stores report at <https://localhost:8888/#/reports/stores/local>.
+Encryption status can be seen on the node's stores report, reachable through: `http(s)://nodeaddress:8080/#/reports/stores/local` (or replace `local` with the node ID). For example, if you are running a [local cluster](secure-a-cluster.html), you can see the node's stores report at <https://localhost:8080/#/reports/stores/local>.
```
How do you find the node ID?

sploiselle commented on v19.1/encryption.md:
```diff
@@ -191,7 +223,11 @@ Information about keys is written to [the logs](debug-and-error-logs.html), incl
+
+The user can change the encryption type from plaintext to encryption, between different encryption algorithms
+(using various key sizes), or from encryption to plaintext. When changing the encryption type to plaintext, the data key registry is no longer encrypted and all previous data keys are readable by anyone. All data on the store is effectively readable. When changing from plaintext to encryption, it will take some time for all data to eventually be re-written
+and encrypted.
```
spurious line break

sploiselle commented on v19.1/encryption.md:
```diff
@@ -191,7 +223,11 @@ Information about keys is written to [the logs](debug-and-error-logs.html), incl
+### Changing encryption algorithm or keys
+
+The user can change the encryption type from plaintext to encryption, between different encryption algorithms
+(using various key sizes), or from encryption to plaintext. When changing the encryption type to plaintext, the data key registry is no longer encrypted and all previous data keys are readable by anyone. All data on the store is effectively readable. When changing from plaintext to encryption, it will take some time for all data to eventually be re-written
```
spurious line break

sploiselle commented on v19.1/encryption.md:
```diff
@@ -191,7 +223,11 @@ Information about keys is written to [the logs](debug-and-error-logs.html), incl
 * New key information after data key rotation.
 
-#### Changing encryption algorithm or keys
+### Changing encryption algorithm or keys
```
While this is true at the highest level, I wonder if we could break these situations into something more obvious to users like breaking things out into more sections by including "Enable Encryption" and "Disable Encryption".

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0796eb350680a821110d5c7cda55c134c20413af/

jseldess commented:
___
*[v19.1/authentication.md, line 7 at r2](https://reviewable.io/reviews/cockroachdb/docs/4420#-LZajr0sY3kNrbtXi28D-r2-7:-L_IY4iJ-y31uvIXI_ns:bg04o1p) ([raw file](https://github.com/cockroachdb/docs/blob/e4652a4f1693a6ce52584caf18d80e173daad1d1/v19.1/authentication.md#L7)):*
<details><summary><i>Previously, sploiselle (Sean Loiselle) wrote…</i></summary><blockquote>

I think the prevalence of the term SSL merits the change:

https://trends.google.com/trends/explore?geo=US&q=digital%20certificate,ssl
</blockquote></details>

Wow, good point, Sean. Shouldn't we technically use TLS? Even that is way more prevelant than digital certificate. https://trends.google.com/trends/explore?geo=US&q=digital%20certificate,tls

Amruta, can you open an issue to look into this more and address?  


<!-- Sent from Reviewable.io -->

jseldess commented:
___
*[v19.1/authentication.md, line 7 at r2](https://reviewable.io/reviews/cockroachdb/docs/4420#-LZajr0sY3kNrbtXi28D-r2-7:-L_IYVLnCH8uqGHiS1WV:b-tkoij) ([raw file](https://github.com/cockroachdb/docs/blob/e4652a4f1693a6ce52584caf18d80e173daad1d1/v19.1/authentication.md#L7)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Wow, good point, Sean. Shouldn't we technically use TLS? Even that is way more prevelant than digital certificate. https://trends.google.com/trends/explore?geo=US&q=digital%20certificate,tls

Amruta, can you open an issue to look into this more and address?  
</blockquote></details>

I think these are better comparison:

digital certificate vs ssl certificate: https://trends.google.com/trends/explore?geo=US&q=digital%20certificate,ssl%20certificate

digital certificate vs tls certificate: https://trends.google.com/trends/explore?geo=US&q=digital%20certificates,tls%20certificate


<!-- Sent from Reviewable.io -->

Amruta-Ranade commented:
The certs we use are not SSL certificates; they are TLS certificates. I used the term `digital certs` because it's the term most associated with public key cryptography and independent of the changes in technology (SSL/TLS). In the PKI section, I am discussing digital certs in general and not SSL/TLS certs in particular.

Amruta-Ranade commented:
@jseldess Yes, the PR addresses the issues listed. I had fixed major issues while revamping the security docs; this PR addresses the minor ones (didn't know about these issues back then).

sploiselle commented:
@Amruta-Ranade TIL! I think it would be useful to do _something_ to capture people who are looking for SSL certificates, or explain that all of these types of certificates are commonly referred to as the same thing, though they are in fact different.

sploiselle commented:
Looked into this some more this and realized my confusion centers on the fact that you use these certificates with the Postgres feature `ssl-mode`. So, even though we've inherited an inaptly named feature, I think it's important to do our best to ensure that our documentation aligns with the mental model of our product. It's unintuitive to me, at least, that you use "digital certificates" or "TLS certificates" when the URI parameter is called "ssl-mode".
